app/models/concerns/submit_result.rb

```diff
@@ -0,0 +1,56 @@
+class SubmitResult
+  module SubmitResultType
+    SUCCESSFULL = "successfull"
+    HAS_SYNTAX_ERRORS = "has_syntax_errors"
+    PENDING_REFACTORING = "pending_refactoring"
+  end
```

Comment:
This looks like a collection that can be stored in a yaml file with keys and values. Then you can load what you need from there